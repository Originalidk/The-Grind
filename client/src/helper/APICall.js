// future dev testing
// http://localhost:3001
// Production link
// https://TimeTrekker.onrender.com

export const APICall = "http://localhost:3001";
// export const APICall = "https://TimeTrekker.onrender.com";