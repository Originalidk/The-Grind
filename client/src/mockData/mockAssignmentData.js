const mockAssignmentData = [
  {
    id: 11,
    title: "assignment 1",
    deadline: null,
    recurring: false,
    completed: false,
    frequency: null,
    createdAt: "2023-07-18T04:57:36.000Z",
    updatedAt: "2023-07-18T04:57:36.000Z",
    UserId: 1,
    SectionId: null,
  },

  {
    id: 12,
    title: "assignment 2",
    deadline: null,
    recurring: true,
    completed: false,
    frequency: null,
    createdAt: "2023-07-18T04:57:40.000Z",
    updatedAt: "2023-07-18T04:57:40.000Z",
    UserId: 1,
    SectionId: null,
  },

  {
    id: 13,
    title: "assignment 3",
    deadline: '"2023-07-22',
    recurring: false,
    completed: false,
    frequency: null,
    createdAt: "2023-07-18T04:57:46.000Z",
    updatedAt: "2023-07-18T04:57:46.000Z",
    UserId: 1,
    SectionId: null,
  },

  {
    id: 14,
    title: "assignment 4",
    deadline: '"2023-07-28',
    recurring: true,
    completed: false,
    frequency: null,
    createdAt: "2023-07-18T04:57:53.000Z",
    updatedAt: "2023-07-18T04:57:53.000Z",
    UserId: 1,
    SectionId: null,
  },

  {
    id: 15,
    title: "assignment 5",
    deadline: '"2023-07-20',
    recurring: true,
    completed: false,
    frequency: "Biweekly",
    createdAt: "2023-07-18T04:58:02.000Z",
    updatedAt: "2023-07-18T04:58:02.000Z",
    UserId: 1,
    SectionId: null,
  },

  {
    id: 16,
    title: "assignment 6",
    deadline: '"2023-07-29',
    recurring: true,
    completed: false,
    frequency: "Monthly",
    createdAt: "2023-07-18T04:58:10.000Z",
    updatedAt: "2023-07-18T04:58:10.000Z",
    UserId: 1,
    SectionId: null,
  },

  {
    id: 17,
    title: "assignment 7",
    deadline: '"2023-08-04',
    recurring: false,
    completed: false,
    frequency: null,
    createdAt: "2023-07-18T04:58:17.000Z",
    updatedAt: "2023-07-18T04:58:17.000Z",
    UserId: 1,
    SectionId: null,
  },
];

export default mockAssignmentData;