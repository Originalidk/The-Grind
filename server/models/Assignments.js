const { Sequelize } = require("sequelize");
const moment = require("moment");

// Export anonymous function so that can be accesed in other files
module.exports = (sequelize, DataTypes) => {
    // Creating a table in sequelize
    const Assignments = sequelize.define("Assignments", {
        title: {
            type: DataTypes.STRING,
            allowNull: false,
        },
        description: {
            type: DataTypes.STRING,
            allowNull: false,
        },
        deadline: {
            type: DataTypes.DATEONLY,
            allowNull: false,
            get() {
                return moment(this.getDataValue('deadline')).format('DD.MM.YYYY');
            }
        },
        recurring: {
            type: DataTypes.BOOLEAN,
            allowNull: false,
            defaultValue: false,
        }, 
        completed: {
            type: DataTypes.BOOLEAN,
            allowNull: false,
            defaultValue: false,
        },
    })

    return Assignments;
}
