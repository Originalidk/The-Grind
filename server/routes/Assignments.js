const express = require("express");
const router = express.Router();

// Get Assignments model from the models folder
const { Assignments } = require("../models");

// Get assignments related to a specific user
router.get("/:userId", async (req, res) => {
  const userId = req.params.userId;
  // Create a variable set to all assignments in the table
  const notCompletedListOfAssignments = await Assignments.findAll({
    where: { completed: false, UserId: userId },
  });
  const completedListOfAssignments = await Assignments.findAll({
    where: { completed: true, UserId: userId },
  });
  const listOfAssignments = [
    ...notCompletedListOfAssignments,
    ...completedListOfAssignments,
  ];

  // Return response in json format
  res.json(listOfAssignments);
});

/*
// Get specific assignment
router.get('/updateAssignment/:assignmentId', async (req, res) => {
    // Create a variable set to the assignmentId param in the URL
    const assignmentId = req.params.assignmentId;
    // Get the assignment where id is equal to the id we are finding
    const assignment = await Assignments.findByPK(assignmentId);
    // Return response in json format
    res.json(assignments);
});
*/
// Post an assignment to a specific user
router.post("/", async (req, res) => {
  // Create a variable set to the assignment data
  const assignment = req.body;
  // Add data to the database
  await Assignments.create(assignment);
  // Return response in json format
  res.json(assignment);
});

router.put("/title", async (req, res) => {
  // Create a variable set to the changed data
  const { title, id } = req.body;
  // Create a variable set to the id
  //const id = req.params.id;
  // Update data to the database
  await Assignments.update({ title: title }, { where: { id: id } });
  // Return response in json format
  res.json(req.body);
});

router.put("/desc", async (req, res) => {
  // Create a variable set to the changed data
  const { description, id } = req.body;
  // Create a variable set to the id
  //const id = req.params.id;
  // Update data to the database
  await Assignments.update({ description: description }, { where: { id: id } });
  // Return response in json format
  res.json(req.body);
});

router.put("/deadline", async (req, res) => {
  // Create a variable set to the changed data
  const { deadline, id } = req.body;
  // Create a variable set to the id
  //const id = req.params.id;
  // Update data to the database
  await Assignments.update({ deadline: deadline }, { where: { id: id } });
  // Return response in json format
  res.json(req.body);
});

router.put("/completed", async (req, res) => {
  // Create a variable set to the changed data
  const { completed, id } = req.body;
  const assignmentObj = await Assignments.findOne({ where: { id: id } });
  if (assignmentObj.recurring != null && assignmentObj.recurring === true) {
    // Update data to the database
    await Assignments.update({ completed: completed }, { where: { id: id } });
    res.json("updated");
  } else {
    // else remove it from db
    await Assignments.destroy({ where: { id: id } });
    res.json("deleted");
  }
});

router.put("/resetRecurring", async (req, res) => {
  const listOfAssignments = await Assignments.findAll();
  const filteredList = listOfAssignments.filter(
    (assignment) => assignment.recurring === true
  );
  await Promise.all(
    filteredList.map(async (assignment) => {
      await Assignments.update(
        { completed: false },
        { where: { id: assignment.id } }
      );
    })
  );
  const newListOfAssignments = await Assignments.findAll();
  res.json(newListOfAssignments);
});

router.delete("/:id", async (req, res) => {
  // Create a variable set to the id
  const id = req.params.id;
  // Delete data from the database
  await Assignments.destroy({ where: { id: id } });
  // Return response in json format
  res.json("Deleted");
});

module.exports = router;
