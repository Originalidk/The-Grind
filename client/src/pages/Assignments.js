import React from "react";
import axios from "axios";
import { useEffect, useState, useContext } from "react";
import { AuthContext } from "../helper/AuthContext";
import { useNavigate } from "react-router-dom";
import Box from "@mui/material/Box";
import {
  Button,
  Card,
  Container,
  Divider,
  Stack,
  Typography,
} from "@mui/material";
import { useTheme } from "@mui/material/styles";
import Checkbox from "@mui/material/Checkbox";
import FilterListIcon from "@mui/icons-material/FilterList";
import ToggleButton from "@mui/material/ToggleButton";
import EventRepeatIcon from "@mui/icons-material/EventRepeat";
import DeleteIcon from "@mui/icons-material/Delete";
import AddBoxIcon from "@mui/icons-material/AddBox";
import RestartAltIcon from "@mui/icons-material/RestartAlt";

function Assignments() {
  const [listOfAssignments, setListOfAssignments] = useState([]);
  const [filter, setFilter] = useState("all");
  const { authState } = useContext(AuthContext);
  const theme = useTheme();
  let navigate = useNavigate();

  useEffect(() => {
    axios.get(`http://localhost:3001/assignments`).then((response) => {
      setListOfAssignments(response.data);
      //console.log(response.data[0].title);
    });
  }, []);

  useEffect(() => {
    if (filter === "all") {
      axios.get(`http://localhost:3001/assignments`).then((response) => {
        setListOfAssignments(response.data);
      });
    } else if (filter === "recurring") {
      axios.get(`http://localhost:3001/assignments`).then((response) => {
        const filtered = response.data.filter(
          (assignment) => assignment.recurring === true
        );
        setListOfAssignments(filtered);
      });
    } else if (filter === "nonrecurring") {
      axios.get(`http://localhost:3001/assignments`).then((response) => {
        const filtered = response.data.filter(
          (assignment) => assignment.recurring === false
        );
        setListOfAssignments(filtered);
      });
    } else {
      // By deadline
      axios.get(`http://localhost:3001/assignments`).then((response) => {
        const filtered = response.data.sort((a, b) =>
          a.deadline > b.deadline ? 1 : -1
        );
        const filter1 = response.data.filter(
          (assignment) =>
            assignment.deadline === null ||
            assignment.deadline === "Invalid Date"
        );
        const final = [...filtered, ...filter1];
        setListOfAssignments(final);
      });
    }
  }, [filter]);

  const edit = (option, id, key) => {
    if (key === undefined) {
      key = 0;
    }
    if (option === "title") {
      let newTitle = prompt("Enter new title");
      axios
        .put("http://localhost:3001/assignments/title", {
          title: newTitle,
          id: id,
        })
        .then((response) => {
          console.log(response);
          const temp = [...listOfAssignments];
          temp[key] = { ...listOfAssignments[key], title: newTitle };
          setListOfAssignments(temp);
        });
    } else if (option === "description") {
      let newDesc = prompt("Enter new description");
      axios
        .put("http://localhost:3001/assignments/desc", {
          description: newDesc,
          id: id,
        })
        .then((response) => {
          console.log(response);
          const temp = [...listOfAssignments];
          temp[key] = { ...listOfAssignments[key], description: newDesc };
          setListOfAssignments(temp);
        });
    } else {
      let newDeadline = prompt("Enter new deadline (YYYY-MM-DD)");
      axios
        .put("http://localhost:3001/assignments/deadline", {
          deadline: newDeadline,
          id: id,
        })
        .then((response) => {
          console.log(response);
          const temp = [...listOfAssignments];
          temp[key] = { ...listOfAssignments[key], deadline: newDeadline };
          setListOfAssignments(temp);
        });
    }
  };

  const handleCheckboxClick = (id, key) => {
    // find the assignment with the id
    const assignment = listOfAssignments.find(
      (assignment) => assignment.id === id
    );
    // change the completed status
    axios
      .put("http://localhost:3001/assignments/completed", {
        id: id,
        completed: !assignment.completed,
      })
      .then((response) => {
        if (response.data === "updated") {
          const temp = [...listOfAssignments];
          temp[key] = {
            ...listOfAssignments[key],
            completed: !listOfAssignments[key].completed,
          };
          setListOfAssignments(temp);
          setTimeout(() => {
            setListOfAssignments(
              listOfAssignments.filter((val) => {
                return val.id !== id;
              })
            );
          }, 300);
        } else { // deleted
          const temp = [...listOfAssignments];
          temp[key] = {
            ...listOfAssignments[key],
            completed: !listOfAssignments[key].completed,
          };
          setListOfAssignments(temp);
          setTimeout(() => {
            setListOfAssignments(
              listOfAssignments.filter((val) => {
                return val.id !== id;
              })
            );
          }, 300);
        }
      });
  };

  const handleDelete = (id) => {
    axios.delete(`http://localhost:3001/assignments/${id}`).then((response) => {
      setListOfAssignments(
        listOfAssignments.filter((val) => {
          return val.id !== id;
        })
      );
    });
  };

  const onFilterClick = () => {
    if (filter === "all") {
      setFilter("recurring");
    } else if (filter === "recurring") {
      setFilter("nonrecurring");
    } else if (filter === "nonrecurring") {
      setFilter("deadline");
    } else {
      setFilter("all");
    }
  };

  const handleResetRecurringClick = () => {
    axios.put("http://localhost:3001/assignments/resetRecurring").then((response) => {
        setListOfAssignments(response.data);
    });
  };

  return (
    <Box
      sx={{
        display: "flex",
        flexDirection: "column",
        width: "78vw",
      }}
    >
      <Box
        sx={{
          display: "flex",
          flexDirection: "row",
          alignItems: "center",
          py: 1,
          flexGrow: 1,
          mb: 0.5,
        }}
      >
        <Typography
          variant="h4"
          component="h4"
          sx={{ flexGrow: 1, color: theme.palette.primary.main }}
        >
          Assignments
        </Typography>
        <ToggleButton
          onClick={() => {
            navigate(`/newAssignment`);
          }}
        >
          <Typography
            variant="body1"
            component="label"
            sx={{ mr: 0.5, color: theme.palette.secondary.main }}
          >
            Add Assignment
          </Typography>
          <AddBoxIcon
            fontSize="medium"
            sx={{ color: theme.palette.secondary.main }}
          />
        </ToggleButton>
        <ToggleButton
          value="filter"
          onClick={onFilterClick}
          sx={{ ml: 2, color: theme.palette.secondary.main }}
        >
          <Typography variant="body1" component="h4" sx={{ flexGrow: 1 }}>
            {filter}
          </Typography>
          <FilterListIcon />
        </ToggleButton>
        <ToggleButton
          onClick={handleResetRecurringClick}
          sx={{ ml: 2, color: theme.palette.secondary.main }}
        >
          <Typography variant="body1" component="h4" sx={{ flexGrow: 1 }}>
            Reset Recurring
          </Typography>
          <RestartAltIcon />
        </ToggleButton>
      </Box>
      <Divider />
      <Stack spacing={2} sx={{ mt: 2, flexGrow: 1 }}>
        {listOfAssignments.map((value, key) => (
          <Card
            key={key}
            elevation={3}
            sx={{ display: "flex", alignItems: "center" }}
          >
            <Checkbox
              onClick={() => handleCheckboxClick(value.id, key)}
              checked={value.completed}
            />
            <Box
              sx={{
                padding: 1,
                borderColor: theme.palette.primary.main,
                flexGrow: 1,
                "&:hover": {
                  backgroundColor: "#f5f5f5",
                  cursor: "pointer",
                },
              }}
              onClick={() => {
                edit("title", value.id, key);
              }}
            >
              <Typography
                variant="body1"
                component="div"
                sx={{ fontWeight: "bold" }}
              >
                {value.title}
              </Typography>
            </Box>
            {value.recurring && (
              <Box sx={{ display: "flex", alignItems: "center" }}>
                <EventRepeatIcon />
                <Typography
                  variant="body2"
                  component="div"
                  sx={{ color: theme.palette.primary.main, marginLeft: 1 }}
                >
                  Recurring Task
                </Typography>
              </Box>
            )}
            <Box
              sx={{
                padding: "10px",
                width: "13%",
                ml: 1,
                "&:hover": {
                  backgroundColor: "#f5f5f5",
                  cursor: "pointer",
                },
              }}
              onClick={() => {
                edit("deadline", value.id, key);
              }}
            >
              {value.deadline === null || value.deadline === "Invalid date" ? (
                <Typography variant="body2" component="div">
                  Deadline: None
                </Typography>
              ) : (
                <Typography variant="body2" component="div">
                  Deadline: {value.deadline}
                </Typography>
              )}
            </Box>
            <DeleteIcon
              onClick={() => handleDelete(value.id)}
              sx={{ cursor: "pointer", color: theme.palette.secondary.main }}
            />
          </Card>
        ))}
      </Stack>
    </Box>
  );

  //     return (
  //         <Box sx={{display: "flex", flexDirection: "column", width: "100vh"}}>
  //             <Typography variant="h4" component="h4">
  //                 Assignments
  //             </Typography>
  //             <Divider />
  //             <Stack>
  //             {listOfAssignments.map((value, key) => {
  //                 return (
  //                     <Card elevation={3} sx={{ margin: 1, mt: 1, mb: 2,}}>
  //                         <div className="title" onClick={ () => { edit("title", value.id, key) } }> {value.title} </div>
  //                         <div className="description" onClick={ () => { edit("description", value.id, key) } }> {value.description} </div>
  //                         <div className="deadline" onClick={ () => { edit("deadline", value.id, key) } }> {value.deadline} </div>
  //                     </Card>
  //                 )
  //             })}
  //             </Stack>
  //         </Box>
  //   )
}

export default Assignments;
