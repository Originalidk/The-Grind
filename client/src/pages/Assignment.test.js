import React from "react";
import { fireEvent, render, screen, within } from "@testing-library/react";
import Assignments from "./Assignments";
import { MemoryRouter } from "react-router-dom";
import { AuthContext } from "../helper/AuthContext";
import "@testing-library/jest-dom";
import { rest } from "msw";
import { setupServer } from "msw/node";
import mockAssignmentData from "../mockData/mockAssignmentData";
import { act } from "react-dom/test-utils";
import axios from "axios";

const server = setupServer(
  rest.get("http://localhost:3001/assignments/mockId", (req, res, ctx) => {
    return res(ctx.status(200), ctx.json(mockAssignmentData));
  }),
  // set a default response to all other not handled requests
  rest.all("*", (req, res, ctx) => {
    return res(
      ctx.status(200),
    );
  })

);

// Establish requests interception before all tests
beforeAll(() => server.listen());
// Reset any runtime request handlers we may add during the tests
afterEach(() => server.resetHandlers());
// Clean up after all tests are done
afterAll(() => server.close());

const mockNavigator = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockNavigator,
}));

describe("Assignments component", () => {
  const mockAuthState = {
    authState: {
      username: "testuser",
      id: "mockId",
      status: true,
    },
    setAuthState: jest.fn(),
  };

  const AssignmentComponent = () => {
    return render(
      <MemoryRouter>
        <AuthContext.Provider value={mockAuthState}>
          <Assignments />
        </AuthContext.Provider>
      </MemoryRouter>
    );
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("renders Assignments component and fetches data from db", async () => {
    act(() => {
      AssignmentComponent();
    });
    // Check if the component renders without any errors
    expect(screen.getByText("Assignments")).toBeInTheDocument();
    expect(await screen.findByText("assignment 1")).toBeInTheDocument();
    expect(await screen.findByText("assignment 7")).toBeInTheDocument();
  });

  it("should be able to check off assignments with api calls to the backend to update", async () => {
    act(() => {
      AssignmentComponent();
    });
    const axiosPutSpy = jest.spyOn(axios, "put");
    const assignment1 = await screen.findByText("assignment 1");
    const checkbox = within(screen.getByTestId("11")).getByRole("checkbox");
    expect(assignment1).toBeInTheDocument();
    expect(checkbox).toBeInTheDocument();
    expect(checkbox).not.toBeChecked();
    act(() => {
      fireEvent.click(checkbox);
    });
    expect(axiosPutSpy).toHaveBeenCalledWith(
      "http://localhost:3001/assignments/completed",
      {
        id: 11,
        completed: true,
      }
    );
  });

  it("should be able to delete assignments with api calls to the backend to update", async () => {
    act(() => {
      AssignmentComponent();
    });
    const axiosDeleteSpy = jest.spyOn(axios, "delete");
    const assignment1 = await screen.findByText("assignment 1");
    const deleteButton = screen.getByTestId("delete-11");
    expect(assignment1).toBeInTheDocument();
    expect(deleteButton).toBeInTheDocument();
    act(() => {
      fireEvent.click(deleteButton);
    });
    expect(axiosDeleteSpy).toHaveBeenCalledWith(
      "http://localhost:3001/assignments/11"
    );
  });

  it("should be able to navigate to the add assignment page with the add assignment button", async () => {
    act(() => {
      AssignmentComponent();
    });

    const addAssignmentButton = screen.getByTestId("add-assignment-button");
    expect(addAssignmentButton).toBeInTheDocument();
    act(() => {
      fireEvent.click(addAssignmentButton);
    });
    expect(mockNavigator).toHaveBeenCalledWith("/newAssignment");
  });

  it("should be able to edit assignments with api calls to the backend to update", async () => {
    act(() => {
      AssignmentComponent();
    });
    const promptBoxSpy = jest.spyOn(window, "prompt");

    const editButton = await screen.findByText("assignment 1");
    expect(editButton).toBeInTheDocument();
    act(() => {
      fireEvent.click(editButton);
    });
    expect(promptBoxSpy).toHaveBeenCalled();
  });

  it("should be able to filter assignments by its recurring status", async () => {
    act(() => {
      AssignmentComponent();
    });
    const allFilterButton = await screen.findByText("all");
    expect(allFilterButton).toBeInTheDocument();
    act(() => {
      fireEvent.click(allFilterButton);
    });
    const recurringFilterButton = await screen.findByText("recurring");
    const assignment4 = await screen.findByText("assignment 4");
    expect(recurringFilterButton).toBeInTheDocument();
    expect(assignment4).toBeInTheDocument();
    act(() => {
      fireEvent.click(recurringFilterButton);
    });
    const nonRecurringFilterButton = await screen.findByText("nonrecurring");
    expect(nonRecurringFilterButton).toBeInTheDocument();
    const assignment1 = await screen.findByText("assignment 1");
    expect(assignment1).toBeInTheDocument();
  });

  it("should be able to filter assignments by due date", async () => {
    act(() => {
      AssignmentComponent();
    });
    const allFilterButton = await screen.findByText("all");
    expect(allFilterButton).toBeInTheDocument();
    act(() => {
      fireEvent.click(allFilterButton);
    });
    const recurringFilterButton = await screen.findByText("recurring");
    expect(recurringFilterButton).toBeInTheDocument();
    act(() => {
      fireEvent.click(recurringFilterButton);
    });
    const nonRecurringFilterButton = await screen.findByText("nonrecurring");
    expect(nonRecurringFilterButton).toBeInTheDocument();
    act(() => {
      fireEvent.click(nonRecurringFilterButton);
    });
    const deadlineFilterButton = await screen.findByText("deadline");
    expect(deadlineFilterButton).toBeInTheDocument();
    act(() => {
      fireEvent.click(deadlineFilterButton);
    });
    const assignmentPage = screen.findByTestId("assignment-page");
    expect(assignmentPage).toMatchSnapshot("assignmentPage");
  });

  it("should be able to reset recurring assignments", async () => {
    act(() => {
      AssignmentComponent();
    });
    const axiosPutSpy = jest.spyOn(axios, "put");
    const assignment2 = await screen.findByText("assignment 2");
    const checkbox = within(screen.getByTestId("12")).getByRole("checkbox");
    expect(assignment2).toBeInTheDocument();
    expect(checkbox).toBeInTheDocument();
    expect(checkbox).not.toBeChecked();
    act(() => {
      fireEvent.click(checkbox);
    });
    expect(axiosPutSpy).toHaveBeenCalledWith(
      "http://localhost:3001/assignments/completed",
      {
        id: 12,
        completed: true,
      }
    );
    const resetRecurringButton = await screen.findByText("Reset Recurring");
    act(() => {
      fireEvent.click(resetRecurringButton);
    });
    expect(axiosPutSpy).toHaveBeenCalledWith(
      "http://localhost:3001/assignments/resetRecurring/mockId"
    );
  });
});
