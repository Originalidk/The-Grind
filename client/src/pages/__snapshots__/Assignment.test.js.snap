// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Assignments component should be able to filter assignments by due date: assignmentPage 1`] = `Promise {}`;
